weaviate/collection/grpc.py

```diff
@@ -325,6 +341,48 @@ def __call(self) -> List[GrpcResult]:
         except grpc.RpcError as e:
             raise WeaviateGRPCException(e.details())
 
+    def __extract_filters(self, weav_filter: _Filters) -> Optional[weaviate_pb2.Filters]:
+        if weav_filter is None:
+            return None
+        from google.protobuf.timestamp_pb2 import Timestamp
+
+        if isinstance(weav_filter, _FilterValue):
+            timestamp = Timestamp()
+
+            if isinstance(weav_filter.value, datetime.date):
+                timestamp.FromDatetime(weav_filter.value)
```

Comment:
When inserting with `datetime` through the REST API, the object is converted to an ISO string with `T` separation and `microsecond` precision.

Is that format the same as this `Timestamp` object? If not, is there business logic in the gRPC connection on the server to convert the protobuf timestamp back into a format comparable to the stored string?

~~Also, shall we maintain the convention of enforcing UTC `tzinfo` in the absence of a user-defined specification? Is that handled automatically by `Timestamp()`?~~ `.FromDatetime` does assume UTC in the absence of a defined `tzinfo`.

Reply:
> Is that format the same as this Timestamp object? If not, is there business logic in the gRPC connection on the server to convert the protobuf timestamp back into a format comparable to the stored string?

that is a good question, I think not yet. The date filtering does not work yet, but I'd handle that in a separate PR later

Reply:
Okay, makes sense! LGTM :)